### ISSUE TITLE
Datepicker: accessibility issues

### ISSUE THREAD

**Issue by hvinh** [ng-bootstrap/ng-bootstrap#1946]
### Bug description:
Accessibility issue
We noticed the following issues on the date picker widget:
- For the next/previous month buttons: there is no alternative text for screen readers and no tooltip on mouse over for the purpose of these buttons
- The days of the week are read in short format by screen readers: "Mon" instead of "Monday"
- Only the number is read by screen readers for the dates in the calendar: "1" instead of "Monday 1 September"
- After selecting a date, the selected date should be conveyed to screen reader users. This could be achieved by setting the focus back to the input field as raised in #1934 

Tested with NVDA/Firefox and JAWS/IE

### Link to minimally-working plunker that reproduces the issue:
Date picker in a pop-up
https://ng-bootstrap.github.io/#/components/datepicker/examples

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: angular 4.4.6

ng-bootstrap: 1.0.0-beta.5

Bootstrap: 4.0.0-beta.2

**Comment by maxokorokov** [comment]
Also see #1716, as not possible to focus navigation part of the datepicker currently

**Comment by mendeza** [comment]
Driving the Datepicker via keyboard-only - an A11y requirement - is not possible with this component.

**Comment by maxokorokov** [comment]
@mendeza it is possible to use datepicker with keyboard only, please see the [keymap service](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/datepicker/datepicker-keymap-service.ts) for available keys. Screen readers and appropriate aria attributes are not working yet though.

**Comment by maxokorokov** [comment]
The remaining checklist to close is:
- [x] allow to focus items inside the datepicker (#1716)
- [ ] accessibility attributes for calendar (#2319)
- [ ] generic focus trap service (#2322)
- [ ] accessibility attributes for navigation (no PR yet)
- [ ] focus trap for datepicker input (no PR yet)

**Comment by pkozlowski-opensource** [comment]
Seems like everything is done here 🍾
